sdk/src/main/java/com/google/cloud/dataflow/sdk/io/TextIO.java

```diff
@@ -1014,19 +1100,49 @@ private TextWriteOperation(TextSink<T> sink, Coder<T> coder) {
     private static class TextWriter<T> extends FileBasedWriter<T> {
       private static final byte[] NEWLINE = "\n".getBytes(StandardCharsets.UTF_8);
       private final Coder<T> coder;
+      private final String header;
+      private final String footer;
       private OutputStream out;
 
       public TextWriter(FileBasedWriteOperation<T> writeOperation, Coder<T> coder) {
```

Comment:
why do we need multiple constructors for this private class? Should just have one.